src/Symfony/Component/Messenger/Transport/MemoryTransport.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport;
+
+use Symfony\Component\Messenger\Asynchronous\Transport\ReceivedMessage;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+class MemoryTransport implements TransportInterface
+{
+    private $envelopes = array();
+    private $bus;
+
+    public function __construct(MessageBusInterface $bus)
+    {
+        $this->bus = $bus;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function receive(callable $handler): void
+    {
+        foreach ($this->envelopes as $envelope) {
+            $handler($envelope);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function stop(): void
+    {
+        $this->envelopes = array();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function send(Envelope $envelope)
+    {
+        $this->envelopes[] = $envelope;
+    }
+
+    public function onKernelTerminate(): void
```

Comment:
I'd name this `flush` as the fact it's called on `kernel.terminate` doesn't have to leak here.